src/crimson/os/seastore/backref/backref_tree_node.h

```diff
@@ -92,10 +88,6 @@ class BackrefLeafNode
     return TYPE;
   }
 
-  bool may_conflict() const final {
-    return false;
-  }
```

Comment:
This constraint should still be enforced, right?

Reply:
Seems there is no guarantee that trim-alloc and trim-dirty transactions won't conflict on the backref leaf node, I think it should be removed.

Reply:
Hmm, right. That explains why trim-dirty and user-io introduces different level of conflicts to trim-alloc.